packages/gatsby/src/schema/infer/inference-metadata.ts

```diff
@@ -264,9 +336,10 @@ const updateValueDescriptor = (
   }
 
   const delta = operation === `del` ? -1 : 1
-  let typeInfo = descriptor[typeName]
+
+  let typeInfo: ITypeInfo | undefined = descriptor[typeName]
   if (typeInfo === undefined) {
-    typeInfo = descriptor[typeName] = { total: 0 }
+    typeInfo = (descriptor[typeName] as ITypeInfo) = { total: 0 }
```

Comment:
There's too much teling the compiler what to do here, but otherwise it complains about the lack of a value for `empty` in the case it's a string field. I could add a small factory function that returns an `ITypeInfo` matching object based on the `typeName`. Is that extra code to maintain worth it compared to the quick compiler hint that's covered by the tests?

Reply:
Yeah, I think it's OK

Reply:
OK to add the factory function, or to leave it alone?

I've no idea why I've got failing tests now. I can't seem to get master to pass either. :/

Reply:
Leave it as is for now. As for tests - there were some unrelated issues recently. Should be fixed now. Let's wait a bit when they finish.